src/bin/rustc.rs

```diff
@@ -0,0 +1,94 @@
+use std::env;
+
+use cargo::ops::CompileOptions;
+use cargo::ops;
+use cargo::util::important_paths::{find_root_manifest_for_cwd};
+use cargo::util::{CliResult, CliError, Config};
+
+#[derive(RustcDecodable)]
+struct Options {
+    arg_opts: Option<Vec<String>>,
+    flag_package: Option<String>,
+    flag_jobs: Option<u32>,
+    flag_features: Vec<String>,
+    flag_no_default_features: bool,
+    flag_target: Option<String>,
+    flag_manifest_path: Option<String>,
+    flag_verbose: bool,
+    flag_release: bool,
+    flag_lib: bool,
+    flag_bin: Vec<String>,
+    flag_example: Vec<String>,
+    flag_test: Vec<String>,
+    flag_bench: Vec<String>,
+}
+
+pub const USAGE: &'static str = "
+Compile a package and all of its dependencies
+
+Usage:
+    cargo rustc [options] [--] [<opts>...]
+
+Options:
+    -h, --help               Print this message
+    -p SPEC, --package SPEC  The profile to compile for
+    -j N, --jobs N           The number of jobs to run in parallel
+    --lib                    Build only this package's library
+    --bin NAME               Build only the specified binary
+    --example NAME           Build only the specified example
+    --test NAME              Build only the specified test
+    --bench NAME             Build only the specified benchmark
+    --release                Build artifacts in release mode, with optimizations
+    --features FEATURES      Features to compile for the package
+    --no-default-features    Do not compile default features for the package
+    --target TRIPLE          Target triple which compiles will be for
+    --manifest-path PATH     Path to the manifest to fetch depednencies for
+    -v, --verbose            Use verbose output
+
+The <pkgid> specified (defaults to the current package) will have all of its
+dependencies compiled, and then the package itself will be compiled. This
+command requires that a lockfile is available and dependencies have been
+fetched.
```

Comment:
I think that this documentation may want to be touched up a bit, perhaps something along the lines of:

> The specified target for the current package (or package specified by `SPEC` if provided) will be compiled along with all of its dependencies. The specified `<opts>...` will all be passed to the *final* compiler invocation, not any of the dependencies. Note that the compiler will still unconditionally receive arguments such as `-L`, `--extern`, and `--crate-type`, and the specified `<opts>...` will simply be added to the compiler invocation.
> 
> This command requires that only one target is being compiled. If more than one target is available for the current package the filters of `--lib`, `--bin`, etc, must be used to select which target is compiled.